File: algorithms/D/Defuse the Bomb/Defuse the Bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        vector<int>Ans(code.size(),0);
        
        for(int i = 0;i<code.size();i++){
            if (k > 0){
                int total = 0;
                for(int count = 1;count<=k;count++){
                    int num = i + count;
                    num = num % code.size(); 
                    total += code[num];
                }
                Ans[i] = total;
            }
            else if (k < 0){
                int total = 0;
                for(int count = -1;count>=k;count--){
                    int num = i + count;
                    if (num < 0){
                        num = code.size() + num;
                    }
                    total += code[num];
                }
                Ans[i] = total;
            }
            else{
                Ans[i] = 0;
            }
        }
        return Ans;
    }
};

File: algorithms/D/Defuse the Bomb/Defuse the Bomb.java
class Solution {
    public int[] decrypt(int[] code, int k) {
        int[] res = new int[code.length];
        if (k == 0) return res;
        //Define the initial window and initial sum
        int start = 1, end = k, sum = 0;
        if (k < 0) {//If k < 0, the starting point will be end of the array.
            k = -k;
            start = code.length - k;
            end = code.length - 1;
        }
        for (int i = start; i <= end; i++) sum += code[i];
        //Scan through the code array as i moving to the right, update the window sum.
        for (int i = 0; i < code.length; i++) {
            res[i] = sum;
            sum -= code[(start++) % code.length];
            sum += code[(++end) % code.length];
        }
        return res;
    }
}
